benchmark/src/main/java/org/corfudb/benchmark/CorfuTableBenchmark.java

```diff
@@ -0,0 +1,194 @@
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.openjdk.jmh.annotations.Level.Invocation;
+import static org.openjdk.jmh.annotations.Level.Trial;
+
+public class CorfuTableBenchmark {
+
+    @Data
+    public static class Key {
+        @NonNull
+        UUID left;
+
+        @NonNull
+        UUID right;
+
+        public Key(UUID left, UUID right) {
+            this.left = left;
+            this.right = right;
+        }
+    }
+
+    public enum CompositeKeyIndexer implements CorfuTable.IndexSpecification<Key, Key, UUID, Key> {
+        BY_LEFT_OR_RIGHT((k,v) -> ImmutableList.of(k.left, k.right))
+        ;
+
+        CompositeKeyIndexer(CorfuTable.IndexFunction<Key, Key, UUID> indexFunction) {
+            this.indexFunction = indexFunction;
+        }
+
+        final CorfuTable.IndexFunction<Key, Key, UUID> indexFunction;
+
+        @Override
+        public CorfuTable.IndexFunction<Key, Key, UUID> getIndexFunction() {
+            return indexFunction;
+        }
+
+        @Override
+        public CorfuTable.ProjectionFunction<Key, Key, UUID, Key> getProjectionFunction() {
+            return projectionFunction;
+        }
+
+        final CorfuTable.ProjectionFunction<Key, Key, UUID, Key> projectionFunction
+                = (i, s) -> s.map(entry -> entry.getValue());
+    }
+
+    @State(Scope.Benchmark)
+    public static class CorfuTableCommon extends Common {
+
+        CorfuTable<Key, Key, CompositeKeyIndexer, UUID> table;
+
+        List<UUID> leftKeys;
+
+        List<UUID> rightKeys;
+
+        Set<Key> keys;
+
+        double subsetRatio = .05;
+
+        Random rand = new Random();
+
+        @Override
+        public void setup() {
+            super.setup();
+            table = rt.getObjectsView()
+                    .build()
+                    .setTypeToken(
+                            new TypeToken<CorfuTable<Key, Key, CompositeKeyIndexer, UUID>>() {})
+                    .setArguments(CompositeKeyIndexer.class)
+                    .setStreamName(Integer.toString(RND.nextInt()))
+                    .open();
+
+            keys = new HashSet<>();
+            createKeySpace();
+        }
+
+        private void createKeySpace() {
+            leftKeys = new ArrayList<>();
+
+            for (int x = 0; x < num; x++) {
+                leftKeys.add(UUID.randomUUID());
+            }
+
+            // Generate a random subset for the right keys. This is
+            // required to create collisions in the secondary index (i.e
+            // one key in the secondary index can reference multiple values)
+            Collections.shuffle(leftKeys);
+            rightKeys = leftKeys.subList(0, (int) (num * subsetRatio));
+        }
+
+        void write(int batch) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "batch". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)